src/utils.php

```diff
@@ -645,3 +645,87 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Recursively check permissions in a directory and confirm the UID/GID is properly set.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return string|bool Error message or true if all checks pass.
+ */
+function check_permissions_in_directory_recursively( string $dir ) {
+	// Open the directory.
+	if ( $handle = opendir( $dir ) ) {
+		// Loop through each entry in the directory.
+		while ( false !== ( $entry = readdir( $handle ) ) ) {
+			// Skip '.' and '..' directories.
+			if ( $entry === '.' || $entry === '..' ) {
```

Comment:
I'm not sure dots should be skipped: an empty directory with the wrong file permission will still be an issue, this function would not pick it up if is the root one.